lib/stream.js

```diff
@@ -114,14 +113,14 @@ export const getSpawnedResult = async (
 			...stdioStreams.map(stdioStream => waitForStreamEnd(stdioStream, processDone)),
 		]);
 	} catch (error) {
-		return Promise.all([
+		spawned.kill();
+		const results = await Promise.all([
 			{error, signal: error.signal, timedOut: error.timedOut},
 			getBufferedData(spawned.stdout, stdoutPromise),
 			getBufferedData(spawned.stderr, stderrPromise),
 			getBufferedData(spawned.all, allPromise),
 		]);
-	} finally {
-		cleanupStdioStreams(stdioStreams);
-		spawned.kill();
+		cleanupStdioStreams(stdioStreams, error);
```

Comment:
This is only needed on errors. On success, the `end: true` option of `.pipe()` automatically closes those streams.